app/controllers/users/reset_passwords_controller.rb

```diff
@@ -21,22 +21,23 @@ def create
     end
 
     def edit
-      result = PasswordResetTokenValidator.new(token_user).submit
-
-      analytics.password_reset_token(**result.to_h)
-      irs_attempts_api_tracker.forgot_password_email_confirmed(
-        success: result.success?,
-        failure_reason: irs_attempts_api_tracker.parse_failure_reason(result),
-      )
-
-      if result.success?
-        @reset_password_form = ResetPasswordForm.new(build_user)
-        @forbidden_passwords = forbidden_passwords(token_user.email_addresses)
+      if FeatureManagement.redirect_to_clean_edit_password_url?
+        session[:reset_password_token] = params[:reset_password_token]
+        redirect_to change_user_password_url
       else
-        handle_invalid_or_expired_token(result)
+        process_edit_password_page_load
       end
     end
 
+    def change_password
```

Comment:
Do we need to create a new route, or could we just redirect to the same page without the URL parameter?

Reply:
I was not sure which way I wanted to go, I was afraid that it would cause too much work on a single method and having to have the method reload on itself on first load would make it a little confusing code wise.

Reply:
Personally, I find it a bit more confusing to have separate routes and non-conventional actions, since ultimately this is just a single page. Also, `change_password` seems to try to both `render 'edit'` as well as `redirect_to new_user_password_url`, so I'm not really sure what the expected behavior there is meant to be.

Maybe we could pull it out into a `before_action` so that it doesn't have to be managed in the route action method itself?

e.g.

```rb
before_action :validate_reset_password_token, only: :edit

def validate_reset_password_token
  result = PasswordResetTokenValidator.new(token_user).submit

  # ...

  redirect_to new_user_password_url
end
```

Reply:
We would still need to refresh the edit page on the before action to not show the token in the url, even if we validate the token. But I can rework it to reload the edit method and check if its stored in the session or not.